internal/db/subscriptions.go

```diff
@@ -92,10 +92,46 @@ func SubscribeUserToDefaultPlan(ctx context.Context, db *gorm.DB, username strin
 	return SubscribeUserToPlan(ctx, db, user, plan, &model.SubscriptionOptions{})
 }
 
-// GetActiveSubscription retrieves the user plan record that is currently active for the user. The effective start
-// date must be before the current date and the effective end date must either be null or after the current date.
-// If multiple active user plans exist, the one with the most recent effective start date is used. If no active
-// user plans exist for the user then a new one for the basic plan is created.
+// GetActiveSubscriptionForDate retrieves information about the subscription active on the specified date. For a
+// subscription to be returned by this function, its effective start date must be beore the specified date and its
+// effective end date must be after the specified date. If multiple subscriptions are active on the specified date then
+// the one with the most redent effective start date is used.
+func GetActiveSubscriptionForDate(
+	ctx context.Context,
+	db *gorm.DB,
+	username string,
+	date time.Time,
+) (*model.Subscription, error) {
+	wrapMsg := fmt.Sprintf("unable to get the active user plan at %s", date)
+	var err error
+
+	// Look up the subscription.
+	var subscription model.Subscription
+	err = db.
+		WithContext(ctx).
+		Table("subscriptions").
+		Joins("JOIN users ON subscriptions.user_id=users.id").
+		Where("users.username=?", username).
+		Where(
+			db.Where("? BETWEEN subscriptions.effective_start_date AND subscriptions.effective_end_date", date).
+				Or("? > subscriptions.effective_start_date AND subscriptions.effecrtive_end_date IS NULL"),
```

Comment:
`effecrtive` is a typo here, I suspect

Reply:
Ah, yes, that is a typo. Thanks for catching that.